### PR TITLE
Update to latest JavaScriptKit

### PR DESCRIPTION
**Pull request by j-f1**
Pending the merger of https://github.com/swiftwasm/JavaScriptKit/pull/26, everything needed for the output of `webidl2swift` to work will be included in the shipping version of JavaScriptKit! I made several changes to align the APIs and fixed a bug where callback protocols could not be passed as a function.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
Hi @Unkaputtbar, @PSchmiedmayer, would you be able to have a look at this PR at some point? Otherwise, if you need any help in maintaining the project, please let me know. The SwiftWasm team would be happy to help, as it became quite important for us while developing [DOMKit](https://github.com/swiftwasm/DOMKit).

File: Sources/WebIDL/IntermediateRepresentation/Member Nodes/MethodNode.swift [hunk 1]
```diff
@@ -20,16 +20,72 @@ class MethodNode: MemberNode, Equatable {
         true
     }
 
-    // swiftlint:disable cyclomatic_complexity function_body_length
```
Comment (mjburghard) [inline Sources/WebIDL/IntermediateRepresentation/Member Nodes/MethodNode.swift:23]: Add the following comment to disable the lint warning
```suggest
// swiftlint:disable cyclomatic_complexity function_body_length
```

File: Sources/WebIDL/IntermediateRepresentation/Member Nodes/MethodNode.swift [hunk 2]
```diff
@@ -103,29 +163,41 @@ class MethodNode: MemberNode, Equatable {
...
                         if let cNode = dataTypeNode as? ClosureNode {
-                             closureNode = cNode
+                            closureNode = cNode
+                            closureOptional = false
                         } else if let aliasNode = dataTypeNode as? AliasNode, let cNode = aliasNode.aliased as? ClosureNode {
                             closureNode = cNode
+                            closureOptional = false
                         } else if let aliasNode = dataTypeNode as? AliasNode, let optionalNode = aliasNode.aliased as? OptionalNode, let cNode = optionalNode.wrapped.node as? ClosureNode {
                             closureNode = cNode
+                            closureOptional = false // FIXME: or true?
```
Comment (mjburghard) [inline Sources/WebIDL/IntermediateRepresentation/Member Nodes/MethodNode.swift]: I am also not completely sure about this, but I think it should be `true`

**Review by mjburghard — COMMENTED** [review]
Sorry for the long delay. I am pretty busy with other stuff right now. 
I had a look at your changes and think they are fine in general.

I will talk to @PSchmiedmayer to see if we can get one of you merge rights, too.
It's cool that you are using the stuff we built.

**Comment by PSchmiedmayer** [comment]
Hi @j-f1 @MaxDesiatov, 
great to hear that you are using webidl2swift and that it is an important part of the Swift WASM ecosystem!
@Unkaputtbar thanks for taking a look at the PR.
We are happy to accept the help in maintaining the project. I have granted both of you maintenance rights for the repo. 👍

**Comment by MaxDesiatov** [comment]
Fantastic, I very much appreciate it!

File: Sources/WebIDL/IntermediateRepresentation/Member Nodes/MethodNode.swift [hunk 3]
```diff
@@ -21,15 +21,72 @@ class MethodNode: MemberNode, Equatable {
...
+            } else {
+                guard let node = parameter.dataType.node as? ProtocolNode else { continue }
+                unwrapped = node
+            }
+            guard unwrapped.kind == .callback else { continue }
+            for params in overloads {
+                var params = params
+                guard let idx = params.firstIndex(where: { $0.label == parameter.label }) else { continue }
+                let param = params[idx]
+                guard let method = unwrapped.requiredMembers.first(where: { $0.isMethod && !$0.isStatic && !$0.isSubscript }) as? MethodNode else { continue }
```
Comment (MaxDesiatov) [inline Sources/WebIDL/IntermediateRepresentation/Member Nodes/MethodNode.swift]: nit: line length

```suggestion
                guard let method = unwrapped.requiredMembers.first(
                    where: { $0.isMethod && !$0.isStatic && !$0.isSubscript }
                ) as? MethodNode else { continue }
```

**Review by MaxDesiatov — DISMISSED** [review]
IDK if `void` to `undefined` change was really needed (maybe needs an explanatory comment at `case unknown` or some other better suited location?), but other than line length nitpicks this seems legit.

**Comment by j-f1** [comment]
> IDK if `void` to `undefined` change was really needed

They changed the WebIDL spec from using `void` to `undefined`. I’ve added support back for `void` since a bunch of specs have not been updated yet.

**Review by MaxDesiatov — APPROVED** [review]
LGTM

**Pull request merged**